packages/create-sitecore-jss/src/templates/angular-xmcloud/src/app/jss-link.service.ts

```diff
@@ -39,13 +40,14 @@ export class JssLinkService {
   }
 
   /**
-   * Removes all link elements that match the specified rel attribute.
-   * @param rel - The rel attribute of the links to be removed.
+   * Checks for an existing link element with the same rel and href attributes.
+   * @param headLink - An HTMLLink object to be checked.
+   * @returns {boolean} - True if a matching link exists, false otherwise.
    */
-  removeLinksByRel(rel: string) {
-    const links = this.document.head.querySelectorAll(`link[rel="${rel}"]`);
-    links.forEach((link) => {
-      this.document.head.removeChild(link);
-    });
+  private isDuplicateLink(headLink: HTMLLink): boolean {
+    const existingLink = this.document.querySelector(
```

Comment:
For this we'll be parsing a progressively larger document each time. 
Can this be limited to document.head only? Alternatively, is it possible to prevent duplicate links from being created?

Reply:
Makes sense. Updated.

Yeah unfortunately Angular creates these canonical/accumulating links while navigating to different routes.
Previously, I tried to clear all the link when visiting a new route but unfortunately it even removed the main stylesheet of the app. So now the most we can do we remove duplicate links from the visited routes.